benchmarks/expected-plans/q17.txt

```diff
@@ -1,55 +1,49 @@
+| plan_type     | plan                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
++---------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| logical_plan  | Projection: CAST(SUM(lineitem.l_extendedprice) AS Float64) / Float64(7) AS avg_yearly                                                                                                                                                                                                                                                                                                                                                                                  |
+|               |   Aggregate: groupBy=[[]], aggr=[[SUM(lineitem.l_extendedprice)]]                                                                                                                                                                                                                                                                                                                                                                                                      |
+|               |     Projection: lineitem.l_extendedprice                                                                                                                                                                                                                                                                                                                                                                                                                               |
+|               |       Inner Join: part.p_partkey = __scalar_sq_1.l_partkey Filter: CAST(lineitem.l_quantity AS Decimal128(30, 15)) < CAST(__scalar_sq_1.__value AS Decimal128(30, 15))                                                                                                                                                                                                                                                                                                 |
```

Comment:
This is a better plan because the redundant 
```
Filter: part.p_partkey = lineitem.l_partkey AND lineitem.l_partkey = part.p_partkey
```
, which is already done by the earlier joins, is removed, right?

Reply:
It also pushes the filter

```
|               |       Filter: CAST(lineitem.l_quantity AS Decimal128(30, 15)) < CAST(__scalar_sq_1.__value AS Decimal128(30, 15)) AND __scalar_sq_1.l_partkey = lineitem.l_partkey              |
``

Into the Join which seems like a win to me (avoid generating output)

Reply:
> This is a better plan because the redundant
> 
> ```
> Filter: part.p_partkey = lineitem.l_partkey AND lineitem.l_partkey = part.p_partkey
> ```
> 
> , which is already done by the earlier joins, is removed, right?

Yes, the duplicated filters are removed. Actually why the original plan include duplicate filters is because the `push_down_filter` rule infers additional filters and try to pushdown them down. If they can not be pushed down, those inferred filters are added back to the Filters, this is unnecessary, need to differ the inferred filters and the original filters.